platform/harmony/api.go

```diff
@@ -0,0 +1,99 @@
+package harmony
+
+import (
+	"github.com/spf13/viper"
+	"github.com/trustwallet/blockatlas/coin"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+	"strconv"
+)
+
+type Platform struct {
+	client Client
+}
+
+func (p *Platform) Init() error {
+	p.client = Client{blockatlas.InitClient(viper.GetString("harmony.rpc"))}
+	return nil
+}
+
+func (p *Platform) Coin() coin.Coin {
+	return coin.Coins[coin.ONE]
+}
+
+func (p *Platform) GetTxsByAddress(address string) (blockatlas.TxPage, error) {
+	result, err := p.client.GetTxsOfAddress(address)
+	if err != nil {
+		return []blockatlas.Tx{}, err
+	}
+	return NormalizeTxs(result.Transactions), err
+}
+
+func (p *Platform) CurrentBlockNumber() (int64, error) {
+	return p.client.CurrentBlockNumber()
+}
+
+func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
+	if srcBlock, err := p.client.GetBlockByNumber(num); err == nil {
+		block := p.NormalizeBlock(&srcBlock)
+		return &block, nil
+	} else {
+		return nil, err
+	}
```

Comment:
You don't need this else

```
	if srcBlock, err := p.client.GetBlockByNumber(num); err == nil {
		block := p.NormalizeBlock(&srcBlock)
		return &block, nil
	}
	return nil, err
```

Reply:
I tried to do so, but it seems like that an assignment inside a conditional will make the assigned values live only in the conditional scope. If I write `err` outside the conditional I will get it as `Unresolved reference`, that's why I found the else way to be helpful in that case.

Another way could be declaring a `var err error` before the conditional like so

```go
var err error
if srcBlock, err := p.client.GetBlockByNumber(num); err == nil {
    block := p.NormalizeBlock(&srcBlock)
    return &block, nil
}
return nil, err
```

Reply:
try that

```
srcBlock, err := p.client.GetBlockByNumber(num)
if err != nil {
    return nil, err
}
block := p.NormalizeBlock(&srcBlock)
return &block, nil
```